src/pages/LEVANTE/RegisterUsers.vue

```diff
@@ -155,6 +155,35 @@ const allFields = [
     dataType: 'string',
   },
 ];
+// Used to remove all other fields in the uploaded CSV
+const allFieldsIdsList = allFields.map((a) => a.field);
+const sanitizeAndValidateSubmittedUsers = (users) => {
+  const userIdMap = {};
+  const filteredFieldsUsers = users.map((user) => {
+    const acc = {};
+    Object.keys(user).forEach((uKey) => {
+      if (allFieldsIdsList.includes(uKey)) {
+        acc[uKey] = user[uKey];
+      }
+    });
+    // Allowing 0 as a valid id
+    const idList = [user['childId'], user['teacherId'], user['parentId']].filter((a) => 0 || !!a);
+    if (idList.length === 0) {
+      throw new Error(`User Row validation:, No Child, Teacher or Parent Id found for: ${JSON.stringify(user)}`);
+    }
+    if (idList.length > 1) {
+      throw new Error(`User Row validation: Mutliple Child, Teacher or Parent Id found for: ${JSON.stringify(user)}`);
+    }
+    if (userIdMap[idList[0]]) {
+      throw new Error(`User Row validation:, Duplicate id present for ${JSON.stringify(user)}, with id: ${idList[0]}`);
+    } else {
+      userIdMap[idList[0]] = true;
+    }
+    return acc;
+  });
+
+  return filteredFieldsUsers;
+};
```

Comment:
This is great. Could you change the error handling to add the row/user to the error table and pass in the message there? Additionally, we want the toast notification message to be more informative rather than hold the actual error values. So maybe an error would be something like "Multiple ids found for some row(s)". Then the error table would display the rows/users with the problems. See the example below.
![Screenshot 2024-05-29 at 4 58 40 PM](https://github.com/yeatmanlab/roar-dashboard/assets/80180474/a6474872-4042-4fc3-923e-0cd619bb4536)